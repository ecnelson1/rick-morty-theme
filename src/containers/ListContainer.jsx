import React from 'react';
import { useTheme } from '../../state/ThemeProvider';
import CharacterList from '../components/characters/CharacterList';
import {useCharacters} from '../hooks/characters-hook';

const ListContainer = () => {
    const [loading, characters] = useCharacters();
    const theme = useTheme();
    return (
        <div>
            {loading ? (<h1>LOADING...</h1>)
            : (<CharacterList characters = {characters}/>)}
        </div>
    );
};
export default ListContainer
