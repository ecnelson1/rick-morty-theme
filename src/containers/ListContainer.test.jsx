/**
 * @jest-environment jsdom
 */

import React from 'react';
import {render, screen, waitFor} from '@testing-library/react';
import {rest} from 'msw';
import {setupServer} from 'msw/node';
import ListContainer from './ListContainer';
import mockData from  '../../fixtures/mock-data.json';


const server = setupServer(
    rest.get('https://rickandmortyapi.com/api/character', (req, res, ctx) => {
        return res(ctx.json(mockData));
    })
);

describe('ListContainer', () => {
    beforeAll(() => server.listen());
    afterAll(() => server.close());
    
    it('renders a loading element', async () => {
        render(<ListContainer/>);
        screen.getByText('LOADING...');
    });

    it('renders a list of characters from Hey Arnold', async ()=> {
        render(<ListContainer/>);
        const ul = await screen.findByRole('list', {name: 'character-list'});
        return waitFor(()=> {
        expect(ul).toMatchSnapshot();
        });
    });
});
