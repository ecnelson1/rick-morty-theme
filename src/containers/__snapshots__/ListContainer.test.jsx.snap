// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContainer renders a list of characters from Hey Arnold 1`] = `
<ul
  aria-label="character-list"
>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Rick Sanchez"
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
      <p>
        Nme: 
        Rick Sanchez
      </p>
      <p>
        Status: 
        Alive
      </p>
      <p>
        Species: 
        Human
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Morty Smith"
        src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
      />
      <p>
        Nme: 
        Morty Smith
      </p>
      <p>
        Status: 
        Alive
      </p>
      <p>
        Species: 
        Human
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Summer Smith"
        src="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
      />
      <p>
        Nme: 
        Summer Smith
      </p>
      <p>
        Status: 
        Alive
      </p>
      <p>
        Species: 
        Human
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Beth Smith"
        src="https://rickandmortyapi.com/api/character/avatar/4.jpeg"
      />
      <p>
        Nme: 
        Beth Smith
      </p>
      <p>
        Status: 
        Alive
      </p>
      <p>
        Species: 
        Human
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Jerry Smith"
        src="https://rickandmortyapi.com/api/character/avatar/5.jpeg"
      />
      <p>
        Nme: 
        Jerry Smith
      </p>
      <p>
        Status: 
        Alive
      </p>
      <p>
        Species: 
        Human
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Abadango Cluster Princess"
        src="https://rickandmortyapi.com/api/character/avatar/6.jpeg"
      />
      <p>
        Nme: 
        Abadango Cluster Princess
      </p>
      <p>
        Status: 
        Alive
      </p>
      <p>
        Species: 
        Alien
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Abradolf Lincler"
        src="https://rickandmortyapi.com/api/character/avatar/7.jpeg"
      />
      <p>
        Nme: 
        Abradolf Lincler
      </p>
      <p>
        Status: 
        unknown
      </p>
      <p>
        Species: 
        Human
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Adjudicator Rick"
        src="https://rickandmortyapi.com/api/character/avatar/8.jpeg"
      />
      <p>
        Nme: 
        Adjudicator Rick
      </p>
      <p>
        Status: 
        Dead
      </p>
      <p>
        Species: 
        Human
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Agency Director"
        src="https://rickandmortyapi.com/api/character/avatar/9.jpeg"
      />
      <p>
        Nme: 
        Agency Director
      </p>
      <p>
        Status: 
        Dead
      </p>
      <p>
        Species: 
        Human
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Alan Rails"
        src="https://rickandmortyapi.com/api/character/avatar/10.jpeg"
      />
      <p>
        Nme: 
        Alan Rails
      </p>
      <p>
        Status: 
        Dead
      </p>
      <p>
        Species: 
        Human
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Albert Einstein"
        src="https://rickandmortyapi.com/api/character/avatar/11.jpeg"
      />
      <p>
        Nme: 
        Albert Einstein
      </p>
      <p>
        Status: 
        Dead
      </p>
      <p>
        Species: 
        Human
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Alexander"
        src="https://rickandmortyapi.com/api/character/avatar/12.jpeg"
      />
      <p>
        Nme: 
        Alexander
      </p>
      <p>
        Status: 
        Dead
      </p>
      <p>
        Species: 
        Human
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Alien Googah"
        src="https://rickandmortyapi.com/api/character/avatar/13.jpeg"
      />
      <p>
        Nme: 
        Alien Googah
      </p>
      <p>
        Status: 
        unknown
      </p>
      <p>
        Species: 
        Alien
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Alien Morty"
        src="https://rickandmortyapi.com/api/character/avatar/14.jpeg"
      />
      <p>
        Nme: 
        Alien Morty
      </p>
      <p>
        Status: 
        unknown
      </p>
      <p>
        Species: 
        Alien
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Alien Rick"
        src="https://rickandmortyapi.com/api/character/avatar/15.jpeg"
      />
      <p>
        Nme: 
        Alien Rick
      </p>
      <p>
        Status: 
        unknown
      </p>
      <p>
        Species: 
        Alien
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Amish Cyborg"
        src="https://rickandmortyapi.com/api/character/avatar/16.jpeg"
      />
      <p>
        Nme: 
        Amish Cyborg
      </p>
      <p>
        Status: 
        Dead
      </p>
      <p>
        Species: 
        Alien
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Annie"
        src="https://rickandmortyapi.com/api/character/avatar/17.jpeg"
      />
      <p>
        Nme: 
        Annie
      </p>
      <p>
        Status: 
        Alive
      </p>
      <p>
        Species: 
        Human
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Antenna Morty"
        src="https://rickandmortyapi.com/api/character/avatar/18.jpeg"
      />
      <p>
        Nme: 
        Antenna Morty
      </p>
      <p>
        Status: 
        Alive
      </p>
      <p>
        Species: 
        Human
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Antenna Rick"
        src="https://rickandmortyapi.com/api/character/avatar/19.jpeg"
      />
      <p>
        Nme: 
        Antenna Rick
      </p>
      <p>
        Status: 
        unknown
      </p>
      <p>
        Species: 
        Human
      </p>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Ants in my Eyes Johnson"
        src="https://rickandmortyapi.com/api/character/avatar/20.jpeg"
      />
      <p>
        Nme: 
        Ants in my Eyes Johnson
      </p>
      <p>
        Status: 
        unknown
      </p>
      <p>
        Species: 
        Human
      </p>
    </figure>
  </li>
</ul>
`;
